media-plugins/adlplug/adlplug-9999.ebuild

```diff
@@ -0,0 +1,126 @@
+# Copyright 2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake-utils multibuild git-r3
+
+DESCRIPTION="FM synthesizer plugins based on OPL3 and OPN2 sound chip emulations"
+HOMEPAGE="https://github.com/jpcima/ADLplug"
+
+EGIT_REPO_URI="https://github.com/jpcima/${PN}.git"
+if [[ ${PV} == *9999 ]]; then
+	KEYWORDS=""
+else
+	# Upstream release tarballs do not contain the necessary third party libs;
+	# these are available via git submodules per upstream build instructions
+	EGIT_COMMIT="v${PV}"
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="Boost-1.0 BSD GPL-3 GPL-2+ LGPL-3 LGPL-2.1 MIT ISC"
+SLOT="0"
+
+# Upstream offers a VST3 build option, but unfortunately JUCE 5.4.1
+# doesn't support VST3 on Linux
+# see https://github.com/WeAreROLI/JUCE/issues/307
+IUSE="+opl3 +opn2 +lv2 +vst jack alsa pch assertions"
+REQUIRED_USE="
+	|| ( opl3 opn2 )
+	|| ( lv2 vst jack alsa )
+"
+
+# JUCE will stop the build if alsa-lib is not present, regardless
+# of the build options
+DEPEND="
+	x11-libs/libXinerama
+	x11-libs/libXext
+	x11-libs/libXrandr
+	x11-libs/libXcursor
+	media-libs/freetype
+	media-libs/alsa-lib
+	jack? ( virtual/jack )
+"
+RDEPEND="${DEPEND}"
+BDEPEND="
+	dev-util/cmake
+	|| (
+		>=sys-devel/gcc-4.8.1
+		>=sys-devel/clang-3.3
+	)
+	app-text/dos2unix
+"
+
+src_prepare() {
+	MULTIBUILD_VARIANTS=()
+	use opl3 && MULTIBUILD_VARIANTS+=( opl3 )
+	use opn2 && MULTIBUILD_VARIANTS+=( opn2 )
+
+	CMAKE_USE_DIR=${WORKDIR}/${P}
+	BUILD_DIR=${CMAKE_USE_DIR}/build
+
+	# see https://github.com/jpcima/ensemble-chorus/issues/10
+	cd ${CMAKE_USE_DIR}/thirdparty/JUCE
+	dos2unix "modules/juce_graphics/colour/juce_PixelFormats.h"
```

Comment:
It seems like the `dos2unix` invocation here is only required to make the patch work. Is that correct? If so I think it's cleaner to fix the patch directly instead of running `dos2unix` first.

Reply:
When I wrote this, I was concerned about git's autocrlf behavior changing the line endings of the patch during fetching/unpacking, but I realized that was only an issue because of my personal gitconfig. I've fixed the patch and taken out the `dos2unix` stuff.